# 第 11 章 · TD(0)

bootstrap：用 V(s') 更新 V(s)。

配套交互演示：[章节网页](../ch11.html)

## 本节目标

- TD target
- 一步更新

## 1. TD 更新

不需等 episode 结束：用 r + γV(s') 作 target，逐步修正 V(s)。

**TD(0)**

```text
V(s) <- V(s) + alpha * (r + gamma V(s') - V(s))
```

In [1]:
import sys
from pathlib import Path
ROOT = Path.cwd()
if not (ROOT / "labs").exists() and (ROOT.parent / "labs").exists():
    ROOT = ROOT.parent
if str(ROOT) not in sys.path:
    sys.path.insert(0, str(ROOT))
sys.path.insert(0, str(ROOT / "labs"))
sys.path.insert(0, str(ROOT / "labs" / "ch11"))
import matplotlib.pyplot as plt
from common.mpl_setup import configure_matplotlib
configure_matplotlib()
from IPython.display import display, Image
from rl import *

In [2]:
td_demo()

TD(0): V=2.0  target=4.60  V'=2.52 (网页 2.52)


**思考** · target 约多少？

<details><summary>查看答案</summary>

r + γV(s') = 1 + 0.9×4 = 4.6。

</details>

## 小结

TD 在线 bootstrap。对照 [ch11.html](../ch11.html)。

## 练习

1. α 过大？
2. 与 MC 区别？